triangle/example.java

```diff
@@ -0,0 +1,53 @@
+public class Triangle {
+
+    private double side1;
+    private double side2;
+    private double side3;
+
+    public Triangle(double side1, double side2, double side3) throws TriangleException {
+        this.side1 = side1;
+        this.side2 = side2;
+        this.side3 = side3;
+
+        if (allSidesAreZero() || hasImpossibleSides() || violatesTriangleInequality()) {
+            throw new TriangleException();
+        }
+    }
+
+    public TriangleKind getKind() {
```

Comment:
Might make sense to embed the `TriangleKind` enum directly in this class and just call it `Kind`.